### PR TITLE
Add `bv{Zero,One}` helpers, hlint config

### LINKED ISSUE #257: Helpers for constructing bitvectors 0 and 1
It's common to construct literal bitvectors representing 0 or 1:

- https://github.com/search?q=repo%3AGaloisInc%2Fwhat4%20%2FbvLit.%2Bzero%2F&type=code
- https://github.com/search?q=repo%3AGaloisInc%2Fcrucible%20%2FbvLit.%2Bzero%2F&type=code
- https://github.com/search?q=repo%3AGaloisInc%2Fwhat4+%2FbvLit.%2Bone%2F&type=code
- https://github.com/search?q=repo%3AGaloisInc%2Fcrucible+%2FbvLit.%2Bone%2F&type=code

The best way to do that in the current interface is:
```haskell
bvLit sym w (BV.zero w)
bvLit sym w (BV.one w)
```
This is a bit of a pain, since you have to repeat the width repr, and to use functions from two separate packages. Instead, it might be nice to export
```
bvZero, bvOne :: IsSymInterface sym => NatRepr w -> IO (SymBV sym w)
```
The names also make the intent more obvious.

### PR DESCRIPTION
**Pull request by langston-barrett**
Fixes #257. The HLint configuration only checks that these helpers are used where appropriate. I used it to find places where they would be useful. It may also serve as a template for downstream repos. I added HLint checking to CI as well.

### CODE REVIEW

File: what4/src/What4/Interface.hs [hunk 1]
```diff
@@ -3294,3 +3298,16 @@ data Statistics
...
+
+-- | An alias for 'minUnsignedBv'.
+--
+-- Useful in contexts where you want to convey the zero-ness of the value more
+-- than its minimality.
+bvZero :: (1 <= w, IsExprBuilder sym) => sym -> NatRepr w -> IO (SymBV sym w)
+bvZero = minUnsignedBV
+
+bvOne :: (1 <= w, IsExprBuilder sym) => sym -> NatRepr w -> IO (SymBV sym w)
+bvOne sym w = bvLit sym w (BV.one w)
```
Comment (RyanGlScott) [inline what4/src/What4/Interface.hs:3313]: Haddocks?

File: what4/src/What4/Interface.hs [hunk 2]
```diff
@@ -180,6 +180,10 @@ module What4.Interface
     -- * Exceptions
   , InvalidRange(..)
 
+    -- * Bitvector utilities
+  , bvZero
+  , bvOne
```
Comment (RyanGlScott) [inline what4/src/What4/Interface.hs:185]: Perhaps it makes more sense to include this under the "Utility combinators" Haddock section above as a sub-section.

File: .hlint.yaml [hunk 3]
```diff
@@ -1,75 +1,20 @@
...
-- ignore: {name: Use uncurry}
-- ignore: {name: Use unless}
-- ignore: {name: Use unwords}
-- ignore: {name: Use void}
-- ignore: {name: Use when}
+# TODO:
+# - error:
+#     name: Use PP.viaShow
+#     lhs: "Prettyprinter.pretty (show x)"
+#     rhs: 'Prettyprinter.viaShow x'
```
Comment (RyanGlScott) [inline .hlint.yaml:20]: Delete this?

File: what4/src/What4/Interface.hs [hunk 4]
```diff
@@ -1117,7 +1121,7 @@ class ( IsExpr (SymExpr sym), HashableF (SymExpr sym), HashableF (BoundVar sym)
   -- | Return the bitvector of the desired width with all 0 bits;
   --   this is the minimum unsigned integer.
   minUnsignedBV :: (1 <= w) => sym -> NatRepr w -> IO (SymBV sym w)
-  minUnsignedBV sym w = bvLit sym w (BV.zero w)
+  minUnsignedBV sym w = bvZero sym w
```
Comment (RyanGlScott) [inline what4/src/What4/Interface.hs:1124]: Eek, I don't think you meant to change this part! `minUnsigned` and `bvZero` are now defined mutually in terms of each other, so calling either function will result in an infinite loop. (The CI can testify to this.)

I suppose this means that the `.hlint.yaml` file will need to carve out an exception for this module.

Comment (langston-barrett) [reply what4/src/What4/Interface.hs]: Whoops, yep! This was a regex search/replace :sweat:

Comment (langston-barrett) [reply what4/src/What4/Interface.hs]: For some reason, `hlint` doesn't seem to tag this nor `bvOne`. That seems fine, false negatives aren't a big problem for this kind of lint.

File: .hlint.yaml (hunk 3 above)
Comment (langston-barrett) [reply .hlint.yaml]: Yeah, I was thinking we might want to do this in a follow-up, but can remove it for now.

File: what4/src/What4/Interface.hs (hunk 2 above)
Comment (langston-barrett) [reply what4/src/What4/Interface.hs]: Good call

File: what4/src/What4/Interface.hs (hunk 1 above)
Comment (langston-barrett) [reply what4/src/What4/Interface.hs:3314]: For once, I had already written them, I just forgot to push that commit :grin: I'm learning!

**Review by RyanGlScott — APPROVED** [review]

**Pull request merged**